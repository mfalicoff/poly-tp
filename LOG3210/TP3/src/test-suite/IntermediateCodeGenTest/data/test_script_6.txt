Declaration
num a;
bool b;
bool c;
EndDeclaration

script1 {
  while(b) {
    a = 42;
  }
}

script2 {
  while(b) {
    while(c && true) {
      a = 42;
    }
  }
}

script3 {
  if(a > 3) {
    a = 1234;
  }
  else{
    if(b) {
      a = 5678;
    }
  }
}

script1()
script3()
script2()
