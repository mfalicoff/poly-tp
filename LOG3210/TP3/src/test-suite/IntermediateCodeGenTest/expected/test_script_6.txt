_L1
if b == 1 goto _L2
goto _L0
_L2
a = 42
goto _L1
if a > 3 goto _L3
goto _L4
_L3
a = 1234
goto _L0
_L4
if b == 1 goto _L5
goto _L0
_L5
a = 5678
_L6
if b == 1 goto _L7
goto _L0
_L7
_L8
if c == 1 goto _L10
goto _L6
_L10
goto _L9
_L9
a = 42
goto _L8
goto _L6
_L0
